app/services/discussion_service.rb

```diff
@@ -55,7 +55,25 @@ def self.update(discussion:, params:, actor:)
     discussion.save!
 
     ThreadSearchService.index! discussion.id if update_search_vector
-    DiscussionReader.for(discussion: discussion, user: actor).set_volume_as_required!
+    DiscussionReader.for(user: actor, discussion: discussion).set_volume_as_required!
     event
   end
+
+  def self.update_reader(discussion:, params:, actor:)
+    reader = DiscussionReader.for(discussion: discussion, user: actor)
+    actor.ability.authorize! :show, discussion
```

Comment:
I feel like REST gives us a poor API when it looks like this. Exposing the discussion_reader like this feels like we're not abstracting the implementation from the behaviour.

Rather than
/api/v1/discussions/update_reader?{discussion_reader: {starred: val, volume: val}

I'd really like to see the discussion's api documentation look more like this:
/api/v1/discussions/:id/mark_as_read
/api/v1/discussions/:id/set_volume?volume=mute /api/v1/discussions/:id/star
/api/v1/discussions/:id/unstar

Then when we change the way we implement stars our API does not need to change.